ImageAnalysis/image_analysis/base.py

```diff
@@ -199,3 +269,17 @@ def configure(self, **kwargs):
                             value = False
                     setattr(self, key, attr_type(value))
         return self
+
+    def build_input_parameter_dictionary(self) -> dict:
+        """Compiles list of class variables into a dictionary
+
+        Can be overwritten by implementing classes if you prefer more control over the return dictionary.
+        For example, adding units into the key names.
+
+        Returns
+        -------
+        dict
+            A compiled dictionary containing all class variables
+        """
+        self.__dict__.update(vars(self))
```

Comment:
What is this line supposed to do? I think `vars(self)` just returns `self.__dict__` so it just updates itself with its own values.